src/super_context.ml

```diff
@@ -552,40 +551,240 @@ module Scope_key = struct
     sprintf "%s@%s" key (Dune_project.Name.encode scope)
 end
 
-module Action = struct
-  open Build.O
-  module U = Action.Unexpanded
+type targets =
+  | Static of Path.t list
+  | Infer
+  | Alias
+
+module Expander : sig
+  module Resolved_forms : sig
+    type t
+
+    (* Failed resolutions *)
+    val failures : t -> fail list
+
+    (* All "name" for %{lib:name:...}/%{lib-available:name} forms *)
+    val lib_deps : t -> Build.lib_deps
+
+    (* Static deps from %{...} variables. For instance %{exe:...} *)
+    val sdeps    : t -> Path.Set.t
+
+    (* Dynamic deps from %{...} variables. For instance %{read:...} *)
+    val ddeps    : t -> (unit, Value.t list) Build.t String.Map.t
+  end
+
+  type sctx = t
+
+  type expander
+    =  sctx
+    -> dir:Path.t
```

Comment:
Why not pass all these arguments to `with_expander`? This should just be `String_with_vars.Var.t -> Syntax.Version.t -> Value.t list option` to match the signature of the function taken by `Action.Unexpanded.Partial_expand`.

Reply:
Okay, I think that's good as well. Do we want to label those args btw? In pform, I chose the `~syntax_version` and `~pform` labels but I don't have a preference for which one is better. We should do it consistently everywhere however.

Reply:
Not labeling them is fine given that they have different and specific types